platform.md

```diff
@@ -148,6 +155,439 @@ A platform MAY support any number of mixins for a given stack in order to suppor
 Changes introduced by mixins SHOULD be restricted to the addition of operating system software packages that are regularly patched with strictly backwards-compatible security fixes.
 However, mixins MAY consist of any changes that follow the [Compatibility Guarantees](#compatibility-guarantees).
 
+## Lifecycle Interface
+### Platform API Compatibility
+
+The platform SHOULD set `CNB_PLATFORM_API=<platform API version>` in the lifecycle's execution environment
+
+IF `CNB_PLATFORM_API` is set in the lifecycle's execution environment, the lifecycle:
+  - MUST EITHER conform to the matching version of this specification OR
+  - Fail if it does not support `<platform API version>`
+
+### Operations
+
+#### Build
+A single app image build* consists of the following phases:
+
+1. Detection
+1. Analysis
+1. Cache Restoration
+1. Build*
+1. Export
+
+A platform MUST execute these phases EITHER by invoking the following phase-specific lifecycle binaries in order:
+1. `/cnb/lifecycle/detector`
+1. `/cnb/lifecycle/analyzer`
+1. `/cnb/lifecycle/restorer`
+1. `/cnb/lifecycle/builder`
+1. `/cnb/lifecycle/exporter`
+
+OR by executing `/cnb/lifecycle/creator`.
+
+> \* **build** is an overloaded term that refers to both a single phase and the operation comprised of the above phases.
+> The meaning of any particular instance of the word **build** must be assessed in context
```

Comment:
feels so bad. not for 0.3, but should we consider renaming one of the contexts

Reply:
The easier thing to change here is the high level operation: we could simply rename this to "Image Creation" to match the `creator`. We could even make this change in the `0.3` spec (this vocab isn't _encoded_ anywhere in lifecycle). What I struggle with is the "build" feels like the correct word for the operation. It matches `pack build`, the common meaning of that term, and the vocabulary used by similar tools.

I wish we had named `builder` something different but I don't have any great ideas here either:
`buildpacker`?
`invoker`?
`orchestrator`?
`buildpack-runner`?

Reply:
Maybe if we never used the verb version of `builder` in the phase-context we could solve the problem? We could always say "during the builder phase" instead of "during the build phase".

Reply:
Yeah, I agree and I don't want to block `0.3` on it, but I do agree we COULD change it. The verb is ideal, but just has so many implications.